### DISCUSSION TITLE
How do hardcoded settings work?

### DISCUSSION THREAD

**Discussion by rwebb616** [rustdesk/rustdesk#7513] [Q&A]
I created an EXE with the key and host set and it appears to work but only on systems which have not previously had Rustdesk installed on them.  For example I have a client that also uses Rustdesk internally with their own server and it's installed on their workstations.  If I close out of their version (right click the icon and choose exit) then launch my executable it doesn't appear to be connected to my self-hosted server for which the settings are supposed to be hard coded.  

Does it actually hard code anything or does it just put the values into the network settings dialog for you?  

I used github actions to generate a build with both my host and key variables set to my self-hosted machine.
Rich

**Comment by GuillaumeHullin** [comment]
I also tried the GitHub action route for hardcoded settings. Technically speaking, it does hardcode your server and key, meaning they are in the actual executable. However, this is still RustDesk where these settings can be changed. If RustDesk is already installed, I'm not surprised that it doesn't change anything.

If you are trying to set up clients in an automated way (I'm working on that too), maybe some PowerShell could help? https://rustdesk.com/docs/en/self-host/client-deployment/

If you want to limit or block features on the RustDesk client itself, you might want to look at the capabilities of the Pro version. The client generator has some interesting "disable" checkboxes. https://rustdesk.com/docs/en/self-host/client-configuration/#1-custom-client-generator-pro-only
